src/zope/schema/_field.py

```diff
@@ -176,7 +177,11 @@ def fromBytes(self, value):
         return value
 
 
-@implementer(INativeString, IFromUnicode, IFromBytes)
+@implementer(
+    *((INativeString,)
+      #+ IFromUnicode, # Text and Bytes both already implement this
+      + ((IFromBytes,) if PY3 else ()) # Bytes already implements this
+      ))
```

Comment:
This is, uh, ... Wow.

I'm not very happy about the way it looks, and I think it might be good to have a comment explaining why we need to be careful not to repeat redundant interfaces here.

Could we make it cleared by using a helper?

```python
@implementer(nonredundant_interfaces_only(INativeString, IFromUnicode, IFromBytes))
```

with

```python
def nonredundant_interfaces_only(*ifaces):
    result = zope.interface.Declaration()
    for iface in ifaces:
        if all(not existing.isOrExtends(iface) for existing in result):
            result += iface
    return result
```

(untested code, could be totally wrong!)

Reply:
That makes sense. Let me give that a try.